doc/private_endpoint_connection_command_guideline.md

```diff
@@ -0,0 +1,88 @@
+                                      Vault(Storage Account). If specified --vault-name and --name/-n, this should be omitted.
+    --name -n                       : The name of the private endpoint connection associated with the Key
+                                      Vault(Storage Account). Required if --id is not specified.
+    --resource-group -g             : Proceed only if Key Vault(Storage Account) belongs to the specified resource group.
+    --vault-name(--account-name)    : Name of the Key Vault(Storage Account). Required if --id is not specified.
+```
+- The `... private-endpoint-connection show/delete` command should look similar to the following, depending on which features are supported by the service.
+```
+Arguments
+    --id                            : The ID of the private endpoint connection associated with the Key
+                                      Vault(Storage Account). If specified --vault-name and --name/-n, this should be omitted.
+    --name -n                       : The name of the private endpoint connection associated with the Key
+                                      Vault(Storage Account). Required if --id is not specified.
+    --resource-group -g             : Proceed only if Key Vault(Storage Account) belongs to the specified resource group.
+    --vault-name(--account-name)    : Name of the Key Vault(Storage Account). Required if --id is not specified.
+```
+
+#### Private Link Resource
+
+- The parent resource should expose a single command group called `private-link-resource` with one commands: `list`.
+- The `... private-link-resource list` command should look similar to the following, depending on which features are supported by the service.
+
+```
+Arguments
+    --vault-name(--account-name) [Required] : Name of the Key Vault(Storage Account).
+    --resource-group -g     : Proceed only if Key Vault belongs to the specified resource group.
+```
+- The output of `... private-link-resource list` should be a array instead of a dictionary.
+
+## Command Authoring
+
+Storage and keyvault modules both are good examples. Feel free to use them as reference.
+
+*Storage*: [PR Link](https://github.com/Azure/azure-cli/pull/12383)
+
+*Keyvault*: [Command Module](https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/keyvault)
+
+#### Parameters
+We provide a build-in function `parse_proxy_resource_id` to parse private endpoint connection id. It can be used to support the `--id` argument.
+```
+from azure.cli.core.util import parse_proxy_resource_id
+pe_resource_id = "/subscriptions/0000/resourceGroups/clirg/" \
+                 "providers/Microsoft.Network/privateEndpoints/clipe/" \
+                 "privateLinkServiceConnections/peconnection"
+result = parse_proxy_resource_id(pe_resource_id)
+namespace.resource_group = result['resource_group']
+namespace.endpoint = result['clipe']
+namespace.name = result['child_name_1']
```

Comment:
From SDK side, there is no connection_id in parameter list. Should we pop connection _id in namespace?

Reply:
Sure. I can add extra part for this one. Using storage as the reference.